cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/OrderByTest.scala

```diff
@@ -116,11 +118,26 @@ class OrderByTest extends DocumentingTest {
         """MATCH (n)
           |RETURN n.length, n.name, n.age
           |ORDER BY n.length""".stripMargin, ResultAssertions((r) => {
-          r.toList should equal(List(Map("n.age" -> 34, "n.name" -> "A", "n.length" -> 170), Map("n.age" -> 32, "n.name" -> "C", "n.length" -> 185), Map("n.age" -> 34, "n.name" -> "B", "n.length" -> null)))
+          r.toList should equal(List(Map("n.age" -> 34, "n.name" -> "A", "n.length" -> 170), Map("n.age" -> 32, "n.name" -> "C", "n.length" -> 185), Map("n.age" -> 36, "n.name" -> "B", "n.length" -> null)))
         })) {
         p("The nodes are returned sorted by the length property, with a node without that property last.")
         resultTable()
       }
     }
+    section("Ordering in a `WITH` clause", "order-with") {
+      p(
+        """When specifying `ORDER BY` in a `WITH` clause, the immediate next clause is allowed to rely on that ordering.
+          |This is useful, e.g. when the next clause contains an aggregation which should receive the rows in a particular order.
```

Comment:
I don't think it makes sense to mention `RETURN` clauses, since there can't be any immediately following clause. I will use the rest of your suggestion.